### PR TITLE
Roy

### PR DESCRIPTION
**Pull request by royambr**
### Code blockers from running:
1. EDA-> Cat features: Trying to split user_timezone nan (float) value (ignoring missing values).
Solution: nan to empty str.
2. EDA -> Feature Engineering: weekday not exists also the apply usage is wrong.
Solution: calculate weekday and ask if > 5

### Unoptimized code
1. Unnecessary var initialization inside loop (bins=1000).
2.  

## Data Science
1. Using all features without feature selection:
a. Checking for high correlated features.
b. Verify importance of features (maybe perform RFE).
2. Train-test split using KFOLD maybe wrong, you should check if the time has big influence on the target value.
3. The entire EDA done on the entire dataset, there are insights taken from the test set.
4. No seed for the XGBoost model -> no reproducibility.
5. Used ROC-AUC score in imbalanced ds, Precision-Recall AUC is better.
6. You where not using overfitting detector but used the default it may cause overfitting.
7. You should see results in train and validation and see if there are problems of overfitting.
8. EDA missing pairwise multivariate relationship, for example correlation between features especially with target. Also between categorical features and target by bar plots.

### CODE REVIEW

**Comment by royambr** [comment]
## Data Science - continue
9. You used position as a feature, that means that during inference you should provide each sample 48 times for each position. Also this kind of training proceedure is sparse as there will not be data for all positions for all gigs.
10. There maybe features that have extreamly low variance even with one value only that should have been chacked in the EDA. For example: listing_id. -> it makes the dimentionallity large.
11. Maybe help the model train using class weight as the dataset is imbalanced.
12. The predict_proba with 0.5 as threshold is uneccesary you can simply use predict.
13. You better show precision-recall graph to farther pick the best threshold for inference use.
14. No insights from the EDA, and if there are there is no evidence for them in the modeling.

**Comment by royambr** [comment]
## P.S (After the assignment ended I thought of more modeling issues or improvements):
15. To use this model you should use its probabilities and maybe calibrate them (use calibration curve with bins also for evaluation).
16. Also, maybe aggregate and use gig level instead of gig in listing level and perform sliding window to look at the past stats features to predict the future window (e.g., using RNN).

**Comment by royambr** [comment]
in 16 I mean future CTR (mean is_click)
